metaflow/plugins/kfp/kfp.py

```diff
@@ -89,14 +90,17 @@ def __init__(
 
         self._client = kfp.Client(namespace=api_namespace, userid=username, **kwargs)
 
-    def create_run_on_kfp(self, experiment_name: str, run_name: str):
+    def create_run_on_kfp(self, experiment_name: str, run_name: str, pipeline_parameters: dict):
         """
         Creates a new run on KFP using the `kfp.Client()`.
         """
         # TODO: first create KFP Pipeline, then an experiment if provided else default experiment.
         run_pipeline_result = self._client.create_run_from_pipeline_func(
             pipeline_func=self.create_kfp_pipeline_from_flow_graph(),
-            arguments={"datastore_root": DATASTORE_SYSROOT_S3},
+            arguments={
+                "datastore_root": DATASTORE_SYSROOT_S3,
+                "pipeline_parameters": json.dumps(pipeline_parameters)
```

Comment:
Hmm interesting you need to `json` serialize the `pipeline_parameters` here, does this just break if you provide it as a `dict`?

Reply:
KFP PipelineParam only supports string. Json format might be supported, but python dict cast to string by default uses `'` instead of `"` in string, which is not json. 

Attempting `dict` resulted in this error: `Kfp step function: error: argument --pipeline-parameters: invalid loads value: "{'beta': 20, 'alpha': 0.01}"` (https://kubeflow.corp.dev.zg-aip.net/_/pipeline/#/runs/details/54051cf9-b141-45c4-a9d9-e728a9a730fa)